controllers/kubeadmconfig_controller.go

```diff
@@ -111,9 +122,87 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	// Store Config's state, pre-modifications, to allow patching
 	patchConfig := client.MergeFrom(config.DeepCopy())
 
-	// maybe do something with cluster some day
-	// maybe do something interesting here some day
-	config.Status.BootstrapData = []byte("hello world")
+	// if the machine has cluster and or init defined on it then generate the init regular join
+	if config.Spec.InitConfiguration != nil && config.Spec.ClusterConfiguration != nil {
+		// get both of these to strings to pass to the cloud init control plane generator
+		initdata, err := json.Marshal(config.Spec.InitConfiguration)
+		if err != nil {
+			log.Error(err, "failed to marshal init configuration")
+			return ctrl.Result{}, err
+		}
+		clusterdata, err := json.Marshal(config.Spec.ClusterConfiguration)
+		if err != nil {
+			log.Error(err, "failed to marshal cluster configuration")
+			return ctrl.Result{}, err
+		}
+
+		cloudInitData, err := cloudinit.NewInitControlPlane(&cloudinit.ControlPlaneInput{
+			InitConfiguration:    initdata,
+			ClusterConfiguration: clusterdata,
+		})
+		if err != nil {
+			log.Error(err, "failed to generate cloud init for bootstrap control plane")
+			return ctrl.Result{}, err
+		}
+
+		config.Status.BootstrapData = cloudInitData
```

Comment:
I just noticed that BootstrapData is defined as []byte, does this have implications on serialization that we need to worry about?